clustering/server/src/main/java/org/wildfly/clustering/server/dispatcher/ChannelCommandDispatcherFactory.java

```diff
@@ -132,13 +133,8 @@ public Object handle(Message message) throws Exception {
                 if (context == null) return NoSuchService.INSTANCE;
                 @SuppressWarnings("unchecked")
                 Command<Object, Object> command = (Command<Object, Object>) unmarshaller.readObject();
-                Callable<Optional<Object>> task = new Callable<Optional<Object>>() {
-                    @Override
-                    public Optional<Object> call() throws Exception {
-                        // Wrap in an Optional, since command execution might return null
-                        return Optional.ofNullable(command.execute(context.orElse(null)));
-                    }
-                };
+                // Wrap execution result in an Optional, since command execution might return null
+                ExceptionSupplier<Optional<Object>, Exception> task = () -> Optional.ofNullable(command.execute(context.orElse(null)));
```

Comment:
I'm not sure `context.orElse(null)` brings anything as you tested context nullity just above

Reply:
You're confusing a null context with a non-null, but non-present context.  These are completely different things.
A null context means that there is no command dispatcher registered with the given ID.  An non-null, but empty context means that a command dispatcher is registered with the given ID, but it was created with a null context (which is completely valid).  e.g. CommandDispatcher<Void>.

Reply:
Ok, I see the diffrence now.
Maybe this should be put in a comment in the code as it's easy to be quite confused about what "non-null, but non-present context" mean :)

Reply:
No problem - I'll add some comments to clarify.

Reply:
Done.  See: https://github.com/wildfly/wildfly/pull/10678/commits/002ea7eb2b4833b793ec5aa5a6c2db71328fefbe